modules/govuk/manifests/node/s_content_store.pp

```diff
@@ -17,5 +17,12 @@
     include ::hosts::default
     include ::hosts::backend_migration
     include icinga::client::check_pings
+
+    $app_domain = hiera('app_domain')
+
+    govuk_envvar {
+    'PLEK_SERVICE_WHITEHALL_FRONTEND_URI': value  => "https://whitehall-frontend.${app_domain}";
+    'PLEK_SERVICE_WHITEHALL_URI': value  => "https://whitehall.${app_domain}";
+    }
```

Comment:
I'm not quite following this. What value are you expecting without this change for the Whitehall Frontend backend, and what's the desired value?

Reply:
Without this change the value would be <whitehall-frontend|whitehall>.<production|staging>.govuk-internal.digital  after this change the value should be <whitehall-frontend|whitehall>.<production|staging>.publising.service.gov.uk... 
from speaking with Seb, this should be enough to get the backend URL correct if content-store was to update the router-api db

Reply:
Right, thanks, that makes sense.

In my mind, it would be preferable to keep this configuration with the app, rather than the node (so in this file https://github.com/alphagov/govuk-puppet/blob/master/modules/govuk/manifests/apps/content_store.pp ) as that's a bit more specific. At the moment, this change will only affect the "live" content store, not the "draft" content store, as it runs on a different node, whereas I think the Content Store configuration overall should be changed.

One other thing, I believe the Content Store will change the Router/Router API backend information everytime it does something with a relevant bit of content, so if someone publishes something in Whitehall, the routing information will be updated. So in my mind, it's maybe more common than you're considering.

Reply:
right thanks Chris, understood... we forgot completely about the draft-content-store.
Also, the app_domain for both the content_store and draft_content_store will also need to change from (staging).publishing.service.gov.uk to <staging|production>.govuk-internal.digital  ... but that will need to be done at migration time.